components/requests/RequestHeader.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Box,
+  Flex,
+  HStack,
+  Text,
+  Menu,
+  MenuList,
+  MenuItem,
+  MenuButton,
+  Button,
+} from '@chakra-ui/react'; // Chakra UI
+import { ChevronLeftIcon, ChevronDownIcon } from '@chakra-ui/icons'; // Chakra UI icon
+import Link from 'next/link'; // Link
+import RequestStatusBadge from '@components/internal/RequestStatusBadge'; // Request status badge
+import ApproveRequestModal from '@components/requests/modals/ApproveRequestModal'; // Approve button + modal
+import RejectRequestModal from '@components/requests/modals/RejectRequestModal'; // Reject button + modal
+import CompleteRequestModal from '@components/requests/modals/CompleteRequestModal'; // Mark as complete button + modal
+
+type RequestHeaderProps = {
+  readonly isRenewal: boolean;
+  readonly applicationStatus: // TODO: Change this to the enum that we add
+  'COMPLETED' | 'INPROGRESS' | 'PENDING' | 'REJECTED' | 'EXPIRING' | 'EXPIRED' | 'ACTIVE';
+  readonly createdAt: string;
+  readonly applicationProcessingStepsCompleted: number[]; // The format of this will change. For now, 6 steps is complete.
+  readonly onApprove: () => void;
+  readonly onReject: () => void;
+  readonly onComplete: () => void;
+};
+
+export default function RequestHeader({
+  isRenewal,
+  applicationStatus,
+  createdAt,
+  applicationProcessingStepsCompleted,
+  onApprove,
+  onReject,
+  onComplete,
+}: RequestHeaderProps) {
+  const rightButtons = () => {
```

Comment:
Can we rename this to something like `_renderActionButtons`? JSDoc would be nice too